// Copyright 2017, Beeri 15.  All rights reserved.
// Author: Roman Gershman (romange@gmail.com)
//
syntax = "proto2";

package puma;


enum FieldClass {
  DATA = 1;
  ARRAY_SIZE = 2;
  OPTIONAL = 3;
  STRING_LEN = 4;
}

enum DataType {
  NONE    = 0;

  BOOL    = 1;
  INT32   = 2;
  UINT32  = 3;

  INT64   = 4;
  UINT64  = 5;

  DOUBLE  = 6;
  STRING  = 7;
  MESSAGE = 8;
};
